File: leetcode/2-Reverse-Integer/reverse-integer.cs
# Runtime: 40 ms, faster than 100.00% of C# online submissions for Reverse Integer.
# Memory Usage: 13.4 MB, less than 15.84% of C# online submissions for Reverse Integer.
public class Solution {
    public int Reverse(int x) {
        bool minus = false;
            Queue inter = new Queue();
            if (x < 0)
            {
                minus = true;
            }
            while (x != 0)
            {
                if (minus)
                {
                    inter.Enqueue(-(x % 10));
                }
                else
                {
                    inter.Enqueue(x % 10);
                }
                x = x / 10;
            }
            int result = 0;
            while (inter.Count != 0)
            {
                result = result * 10 + (int)inter.Dequeue();
                if (inter.Count == 1)
                {
                    if (result > 214748364)
                    {
                        return 0;
                    }
                    else if (result == 214748364)
                    {
                        int front = (int)inter.Dequeue();
                        if (minus && front >= 8)
                        {
                            return 0;
                        }
                        if (front >= 7)
                        {
                            return 0;
                        }
                        inter.Enqueue(front);
                    }
                }
            }
            if (minus)
            {
                result = -result;
            }
            return result;
    }
}

File: leetcode/2-Reverse-Integer/reverse-integer.go
// Runtime: 4 ms, faster than 100.00% of Go online submissions for Reverse Integer.
// Memory Usage: 2.3 MB, less than 86.36% of Go online submissions for Reverse Integer.
func reverse(x int) int {
	minus := false
	if x < 0{
		minus = true
	}
	var inter []int
	for x != 0{
		if minus{
			inter = append(inter, -(x%10))
		}else{
			inter = append(inter,x%10)
		}
		x=x/10
	}
	var result int
	result = 0
	for i:=0;i < len(inter);i=i+1 {
		result = result * 10 + inter[i]
		if i == len(inter) - 2{
			if result > 214748364 {
				return 0
			}
			if result == 214748364{
				if minus&&inter[i] >= 8{
					return 0
				}
				if inter[i]>=7{
					return 0
				}
			}
		}
	}
	if minus{
		result = -result
	}
	return result
}